### PR TITLE
chore(web): extract-appeal-fund-action

### PR DESCRIPTION
**Pull request by tractorss**
## PR-Codex overview
This PR focuses on enhancing the fund appeal functionality by introducing new types, builders, and hooks for various dispute kits, along with refactoring existing components for improved clarity and performance.

### Detailed summary
- Added `NumberString` type to `types.ts`.
- Introduced new mock accounts in `mocks/index.ts`.
- Created `FundAppealContext` interface in `fundAppeal/context.ts`.
- Updated `useClassicAppealQuery` with new query key.
- Implemented fund appeal builders for classic, gated, and shutter dispute kits.
- Defined `FundAppealParams` interface in `fundAppeal/params.ts`.
- Added `executeFundAppeal` function in `fundAppeal/execute.ts`.
- Refactored `useFundAppeal` hook for better state management.
- Enhanced `Fund` component to utilize new hooks and manage appeal submissions.
- Updated tests for `classicFundAppealBuilder` to cover various scenarios.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

* **New Features**
  * Support for funding appeals across all dispute kit types, a unified builder/dispatcher and an external hook to power appeal UI flows and submissions.
  * New typed context and param shapes plus an execute function to send prepared appeal transactions.

* **Bug Fixes**
  * Improved appeal query cache key to reduce stale/duplicate queries.

* **Tests**
  * Added comprehensive tests for fund-appeal builders covering structure and edge cases.

* **Chores**
  * Updated test mocks and added a NumberString utility type.

### CODE REVIEW

File: web/src/hooks/useFundAppeal.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      const result = await wrapWithToast(executeTxn, publicClient);
+
+      return result;
+    },
+    onSuccess: (res) => {
+      if (!res.status) return;
+      onSuccess?.();
+      queryClient.invalidateQueries({ queryKey: ["useClassicAppealQuery"] });
+    },
```
[the remark this answers is not in the document]
Comment (tractorss) [reply web/src/hooks/useFundAppeal.ts]: It fetches data for all DK, not separately

**Review by kemuru — APPROVED** [review]

**Review by jaybuidl — APPROVED** [review]
